src/expression.js

```diff
@@ -110,6 +111,10 @@ pp.parseMaybeAssign = function(noIn, refDestructuringErrors, afterLeftParse) {
     node.operator = this.value
     node.left = this.type === tt.eq ? this.toAssignable(left) : left
     refDestructuringErrors.shorthandAssign = 0 // reset because shorthand default was used correctly
+    if (parens && (node.left.type === "ObjectPattern" || node.left.type === "ArrayPattern")) {
+      refDestructuringErrors.wrappingParens = node.left.start
+      this.checkAssignmentErrors(refDestructuringErrors, true)
+    }
```

Comment:
Since the check appears to be entirely local, there's no need to go through `refDestructuringErrors` here -- you can simply directly raise the error.